### PR TITLE
fixed ghosts bug on abandoned mines, caption on abandoned mine dialog

### PR DESCRIPTION
**Pull request by shprotru**
related issues: #5512, #4628

### CODE REVIEW

File: src/fheroes2/heroes/heroes_action.cpp [hunk 1]
```diff
@@ -2169,17 +2169,18 @@ void ActionToCaptureObject( Heroes & hero, const MP2::MapObjectType objectType,
...
-            if ( resource == Resource::UNKNOWN )
-                Dialog::Message( header, body, Font::BIG, Dialog::OK );
-            else
-                DialogCaptureResourceObject( header, body, resource );
-
             // restore the abandoned mine
             if ( objectType == MP2::OBJ_ABANDONEDMINE ) {
                 Maps::Tiles::UpdateAbandonedMineSprite( tile );
                 hero.SetMapsObject( MP2::OBJ_MINES );
+                Interface::Basic::Get().Redraw( Interface::REDRAW_GAMEAREA );
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes_action.cpp:2176]: Could you please verify how it works in "no interface" mode?

Comment (shprotru) [reply src/fheroes2/heroes/heroes_action.cpp:2176]: seems it work same as in `common interface` mode, do you see some differ?

**Comment by ihhub** [comment]
Hi @Branikolog , could you please test this pull request?

**Comment by Branikolog** [comment]
Hi, @ihhub !
Window title works well.
But unfortunately, I couldn't reproduce the original bug on my machine... so I cannot test, whether fix work here.

**Comment by shprotru** [comment]
Greetings, @LeHerosInconnu ! Could you please test this pull request?

**Comment by LeHerosInconnu** [comment]
Hello @shprotru,

> Greetings, @LeHerosInconnu ! Could you please test this pull request?

The window title after the gold mine restoration is displayed now.
Great. :)
The update from the abandoned mine to the restored mine sprite seems correct, but the update of the player's color flag above the mine is visible if the combat is played or replayed after an instant combat.

In fheroes2:

https://user-images.githubusercontent.com/43583994/174312315-20630c39-572d-47e3-b251-a5023acd598e.mp4

I can open a different issue if necessary.

**Comment by LeHerosInconnu** [comment]
Hello @shprotru,

> Hello, @LeHerosInconnu ! I've done with flag update for captured abandoned mine, check, please(wait when .exe will be builded by build bots).

Everything works fine.
Good work. :)

**Comment by ihhub** [comment]
@shprotru , thank you so much for this fix!

**Pull request merged**